lib/fizzy/leb128.hpp

```diff
@@ -27,7 +30,13 @@ std::pair<T, const uint8_t*> leb128u_decode(const uint8_t* input)
 }
 
 template <typename T>
-std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input)
+std::pair<T, const uint8_t*> leb128u_decode(bytes_view input)
+{
+    return leb128u_decode<T>(input.data(), input.data() + input.size());
+}
+
+template <typename T>
+std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input, const uint8_t* end)
```

Comment:
Why no `leb128s_decode(bytes_view input)` helper then?

Reply:
There is longer story for it. I'm not sure what should be the final version.

1. The helper `leb128u_decode(bytes_view input)` was added to limit changes in unit tests.
2. Using `bytes_view` has bad side effect that `(bytes{}.data())` still works, because `const uint8_t*` is convertible to `bytes_view`. Although, looks like clang-tidy might have a check for it.
3. So when I reached  `leb128s_decode()` in the very end, I decided to rather rewrite tests.
4. The `const bytes&` is safer than `bytes_view`. I'm currently using that for `parse_expr()` in incoming PR.

Reply:
Maybe put the helper into tests then, if it's used only there.

Reply:
> Maybe put the helper into tests then, if it's used only there.

Done.